frontend/src/employee-frontend/components/messages/MessagesPageContext.tsx

```diff
@@ -0,0 +1,171 @@
+import React, {
+  createContext,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState
+} from 'react'
+import { Loading, Paged, Result } from '../../../lib-common/api'
+import { useRestApi } from '../../../lib-common/utils/useRestApi'
+import { UUID } from '../../types'
+import { getMessageDrafts, getReceivedMessages, getSentMessages } from './api'
+import { DraftContent, MessageThread, SentMessage } from './types'
+import { AccountView } from './types-view'
+
+const PAGE_SIZE = 20
+
+export interface MessagesPageState {
+  selectedDraft: DraftContent | undefined
```

Comment:
It seems that at the moment, while everything else works, the receivers are not preserved correctly when a draft is reopened.

Perhaps the reason is that instead of the actual messaging account ids we should store the `SelectorNode` ids after extracting them with `getSelected(selectedReceivers).map(({value}) => value)`. Reselecting those nodes would then be done by something like
```
recordedIds
  .map(id => ({selectorId: id, selected: true}))
  .reduce((selection, selectorChange) => updateSelector(selection, selectorChange), emptySelection)
```
The downside is that there is no guarantee that the messaging account ids are rendered the same way when the draft is reopened (there could have been changes to a group, for example). This has been discussed before and I think that the consensus was that it still is the way it this is supposed to work.

Reply:
Changed the architecture so that the selected `SelectorNode`s are saved instead of message accounts.